### PR TITLE
Add a system-test

### PR DESCRIPTION
**Pull request by ofrobots**
Second attempt at #487, but travis on-merge tests should pass now.

### CODE REVIEW

File: bin/run-test.sh [hunk 1]
```diff
@@ -50,3 +50,15 @@ fi
 
 # Run non-interference tests
 node test/non-interference/http-e2e.js || exit 1
+
+
+# If on travis, and this is not a pull request build, run the system-tests
+# as well.
+if [ "${TRAVIS_PULL_REQUEST}" = "false" ]; then
```
Comment (matthewloring) [inline bin/run-test.sh]: This will probably fail locally as the travis env variable isn't set but the key + iv variables aren't either.

Comment (ofrobots) [inline bin/run-test.sh]: Yep, but we also don't want it running on appveyor and circle. I guess I have to go find the laundry list of environment variables to add here. :/

Comment (ofrobots) [inline bin/run-test.sh]: Will do this in a follow-on.

Comment (matthewloring) [inline bin/run-test.sh]: SGTM

**Pull request merged**